test/torchaudio_unittest/functional/autograd_impl.py

```diff
@@ -24,7 +24,7 @@ def assert_grad(
         inputs_ = []
         for i in inputs:
             if torch.is_tensor(i):
-                i = i.to(dtype=self.dtype, device=self.device)
+                i = i.to(dtype=torch.cdouble if i.is_complex() else self.dtype, device=self.device)
```

Comment:
`self.complex_dtype` should do.

https://github.com/pytorch/audio/blob/99b5ef5ca0c3a48d8b703d0e1260aafe3334e844/test/torchaudio_unittest/common_utils/case_utils.py#L86-L92